diff --git a/dist/index.js b/dist/index.js
index 8248918ba533c202932e444ca2910b6bfe57be56..83a75eb4b9b5e06e62fd44c8944199ca90e693a1 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -26,15 +26,14 @@ var verifyFirebaseAuth = (userConfig) => {
   };
   const checkRevoked = false;
   return async (c, next) => {
-    const authorization = c.req.raw.headers.get(config.authorizationHeaderKey);
-    if (authorization === null) {
+    const jwt = getCookie(c, 'idToken');
+    if (!jwt) {
       const status = 400;
       throw new HTTPException(status, {
-        res: new Response("Bad Request", { status }),
-        message: "authorization header is empty"
+        res: new Response("Bad Request! Cookies are empty", { status }),
+        message: "Cookies are empty"
       });
     }
-    const jwt = authorization.replace(/Bearer\s+/i, "");
     const auth = Auth.getOrInitialize(
       config.projectId,
       config.keyStore ?? config.keyStoreInitializer(c)
